src/checkbox/checkbox.component.ts

```diff
@@ -178,20 +177,15 @@ export class Checkbox implements ControlValueAccessor, AfterViewInit {
 	 * Allows double binding with the `indeterminateChange` Output.
 	 */
 	@Input() set indeterminate(indeterminate: boolean) {
-		if (indeterminate === this._indeterminate) {
```

Comment:
If I put that back I might need to put `this.inputCheckbox.nativeElement.indeterminate = indeterminate;` above it otherwise it doesn't update the indeterminate state sometimes.. specifically when it's activated/deactivated programmatically like when you select and deselect one row on a table

Reply:
I'll dig deeper into that because I don't see why it shouldn't

Reply:
It's because of https://github.com/IBM/carbon-components-angular/blob/master/src/checkbox/checkbox.component.ts#L399-L403, do you remember why `resetIndeterminate` is needed there?

Reply:
Since with this PR we wouldn't be enforcing indeterminate to toggle to unchecked, I made it so `set checked` does not reset indeterminate state. It currently only resets indeterminate when the user explicitly toggles the checkbox.